Changes to allow secrets to be fetched and inserted into config

btelnes opened the pull request:
- Refactored update.sh
- Minor change to replace.sh
- Handle and validate command line arguments: role, environment, and access token
- Fetch and replace tokens in config with secrets
- Tokenize a portion of server-vars.yml

chenriksson commented on scripts/update.sh:
```diff
@@ -3,60 +3,136 @@
+
+  # Fetch the latest secrets from the private repo via a personal access token
+  cd $OXA_TOOLS_PATH
+  sudo git clone https://$ACCESS_TOKEN@github.com/microsoft/oxa-tools-config.git
```
Instead of removing and re-cloning each time, would just clone on first time. Would also do checkout of master each time. Soon, once we have prod/int environments and release branch, we'll need to account for that.

chenriksson replied:
Would also keep repositories private, and clone oxa-tools-config into /oxa instead of under /oxa/oxa-tools repo path.

btelnes replied:
Made the requested changes,

btelnes commented on config/server-vars.yml:
```diff
@@ -35,12 +35,12 @@ EDXAPP_FEATURES:
+#EDXAPP_EMAIL_HOST: "%%EDXAPP_EMAIL_HOST%%"
+#EDXAPP_EMAIL_PORT: %%EDXAPP_EMAIL_PORT%%
+#EDXAPP_EMAIL_USE_TLS: %%EDXAPP_EMAIL_USE_TLS%%
+EDXAPP_EMAIL_HOST: "%%EDXAPP_EMAIL_HOST%%"
```
Leaving as-is for now.

btelnes commented:
Addressed code review comments.

https://github.com/Microsoft/oxa-tools/pull/6/commits/f664a5b1aebcf3905035b67e4f0c168915efb868

Thanks,
Brian

From: Christy Henriksson [mailto:notifications@github.com]
Sent: Friday, October 14, 2016 12:12 PM
To: Microsoft/oxa-tools <oxa-tools@noreply.github.com>
Cc: Brian Telnes <btelnes@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Microsoft/oxa-tools] Changes to allow secrets to be fetched and inserted into config (#6)


@chenriksson requested changes on this pull request.

@btelnes<https://github.com/btelnes> a few minor comments

________________________________

In config/server-vars.yml<https://github.com/Microsoft/oxa-tools/pull/6#pullrequestreview-4320729>:

> @@ -35,12 +35,12 @@ EDXAPP_FEATURES:

   DISABLE_COURSE_CREATION: true



 # Configure remote SMTP server

-#EDXAPP_EMAIL_HOST_USER: "username"

-#EDXAPP_EMAIL_HOST_PASSWORD: "password"

-#EDXAPP_EMAIL_HOST: "smtp.live.com"

-#EDXAPP_EMAIL_PORT: 25

-#EDXAPP_EMAIL_USE_TLS: true

-EDXAPP_EMAIL_HOST: "openedx-mysql"

+#EDXAPP_EMAIL_HOST_USER: "%%EDXAPP_EMAIL_HOST_USER%%"

+#EDXAPP_EMAIL_HOST_PASSWORD: "%%EDXAPP_EMAIL_HOST_PASSWORD%%"

+#EDXAPP_EMAIL_HOST: "%%EDXAPP_EMAIL_HOST%%"

+#EDXAPP_EMAIL_PORT: %%EDXAPP_EMAIL_PORT%%

+#EDXAPP_EMAIL_USE_TLS: %%EDXAPP_EMAIL_USE_TLS%%

+EDXAPP_EMAIL_HOST: "%%EDXAPP_EMAIL_HOST%%"

Once we separate out fullstack config from scalable (oxa deployments), we should have just EDXAPP_EMAIL_HOST for fullstack and uncommented versions of the others for scalable. Probably fine to leave as-is for now.

________________________________

In scripts/update.sh<https://github.com/Microsoft/oxa-tools/pull/6#pullrequestreview-4320729>:

>

     git clone $CONFIGURATION_REPO $OXA_CONFIG_PATH

     cd $OXA_CONFIG_PATH

     git checkout $CONFIGURATION_VERSION



     pip install -r requirements.txt

   fi

-

+

+  # Remove the oxa-tools-config directory if it exists to avoid the error

+  # "fatal: destination path 'oxa-tools-config' already exists and is not an empty directory."

+  if [[ -d $OXA_TOOLS_PATH/oxa-tools-config ]]; then

+    sudo rm -R $OXA_TOOLS_PATH/oxa-tools-config

+  fi

+

+  # Fetch the latest secrets from the private repo via a personal access token

+  cd $OXA_TOOLS_PATH

+  sudo git clone https://$ACCESS_TOKEN@github.com/microsoft/oxa-tools-config.git

Instead of removing and re-cloning each time, would just clone on first time. Would also do checkout of master each time. Soon, once we have prod/int environments and release branch, we'll need to account for that.

________________________________

In scripts/update.sh<https://github.com/Microsoft/oxa-tools/pull/6#pullrequestreview-4320729>:

>

     git clone $CONFIGURATION_REPO $OXA_CONFIG_PATH

     cd $OXA_CONFIG_PATH

     git checkout $CONFIGURATION_VERSION



     pip install -r requirements.txt

   fi

-

+

+  # Remove the oxa-tools-config directory if it exists to avoid the error

+  # "fatal: destination path 'oxa-tools-config' already exists and is not an empty directory."

+  if [[ -d $OXA_TOOLS_PATH/oxa-tools-config ]]; then

+    sudo rm -R $OXA_TOOLS_PATH/oxa-tools-config

+  fi

+

+  # Fetch the latest secrets from the private repo via a personal access token

+  cd $OXA_TOOLS_PATH

+  sudo git clone https://$ACCESS_TOKEN@github.com/microsoft/oxa-tools-config.git

Would also keep repositories private, and clone oxa-tools-config into /oxa instead of under /oxa/oxa-tools repo path.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/oxa-tools/pull/6#pullrequestreview-4320729>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AUM5T9l8FGGPyGAwwzOJKDn6t7AuubBkks5qz9QOgaJpZM4KXPdy>.

btelnes commented:
Addressed code review comments.

https://github.com/Microsoft/oxa-tools/pull/6/commits/f664a5b1aebcf3905035b67e4f0c168915efb868

Thanks,
Brian


From: Huseyin Koyun [mailto:notifications@github.com]
Sent: Friday, October 14, 2016 10:37 AM
To: Microsoft/oxa-tools <oxa-tools@noreply.github.com>
Cc: Brian Telnes <btelnes@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [Microsoft/oxa-tools] Changes to allow secrets to be fetched and inserted into config (#6)


@ms-hukoyun requested changes on this pull request.

EDXAPP_EMAIL_HOST used twice here. I think one of them should be removed.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/oxa-tools/pull/6#pullrequestreview-4311975>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AUM5T-Ei2ey9U6UsC8S_tykFpJXJ9wgqks5qz73XgaJpZM4KXPdy>.

chenriksson commented on scripts/update.sh:
```diff
@@ -84,18 +85,15 @@ setup() {
-  cd $OXA_TOOLS_PATH
-  sudo git clone https://$ACCESS_TOKEN@github.com/microsoft/oxa-tools-config.git
+    # Fetch the latest secrets from the private repo via a personal access token
+    sudo git clone -b master https://$ACCESS_TOKEN@github.com/microsoft/oxa-tools-config.git
```
Fine for now, but once we add release branch we'll want to separate the clone and checkout commands
